cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/DeclarationHandler.kt

```diff
@@ -405,7 +406,27 @@ open class DeclarationHandler(lang: JavaLanguageFrontend) :
     fun handleEnumConstantDeclaration(
         enumConstDecl: com.github.javaparser.ast.body.EnumConstantDeclaration
     ): EnumConstantDeclaration {
-        return this.newEnumConstantDeclaration(enumConstDecl.nameAsString, rawNode = enumConstDecl)
+        val currentEnum = frontend.scopeManager.currentRecord
+        val result =
+            this.newEnumConstantDeclaration(enumConstDecl.nameAsString, rawNode = enumConstDecl)
+        if (enumConstDecl.arguments.isNotEmpty()) {
+            val arguments =
+                enumConstDecl.arguments.mapNotNull {
+                    frontend.expressionHandler.handle(it)
+                        as? de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
+                }
+            val matchingConstructor =
```

Comment:
Shouldn't this be done by some kind of call resolving in a later stage? But since the java frontend is in dire need of a complete overhaul, I won't block it :)

Reply:
Since the entries of an enum aren't connected by the EOG, they won't be reached by the SymbolResolver if I remember this correctly. That's why I decided to add the target constructor here manually. But this is indeed problematic because the constructor might not have been parsed yet

Reply:
I added a comment, so we remember this in the future